EasyPost/_base/EasyPostClient.cs

```diff
@@ -0,0 +1,150 @@
+        }
+
+        /// <summary>
+        ///     Execute a request against the EasyPost API.
+        /// </summary>
+        /// <typeparam name="T">Type of object to deserialize response data into.</typeparam>
+        /// <returns>An instance of a T type object.</returns>
+        internal async Task<T> Request<T>(Method method, string url, Dictionary<string, object>? parameters = null, string? rootElement = null, ApiVersion? apiVersion = null) where T : class
+        {
+            // Build the request
+            Request request = new Request(url, method, parameters, rootElement, apiVersion);
+            RestRequest restRequest = PrepareRequest(request);
+
+            // Execute the request
+            RestResponse<T> response = await _restClient.ExecuteAsync<T>(restRequest);
+
+            // Check the response's status code
+            if (!response.IsSuccessful)
+            {
+                // HTTP request threw an error (non-2xx response)
+                // RestSharp utilizes .NET HttpStatusCode internally:
+                // https://docs.microsoft.com/en-us/uwp/api/windows.web.http.httpresponsemessage.issuccessstatuscode?view=winrt-22621
+
+                HttpException httpException = HttpException.FromResponse(response);
+                throw httpException;
+            }
+
+            // Prepare the list of root elements to use during deserialization
+            List<string>? rootElements = null;
+            if (request.RootElement != null)
+            {
+                rootElements = new List<string>
+                {
+                    request.RootElement
+                };
+            }
+
+            // Deserialize the response into an object
+            T resource = JsonSerialization.ConvertJsonToObject<T>(response, null, rootElements);
+
+            // Copy this client to the object
+            if (resource is IList list)
+            {
+                foreach (object? element in list)
+                {
+                    (element as EasyPostObject)!.Client = (Client)this;
+                }
+            }
+            else
+            {
+                (resource as EasyPostObject)!.Client = (Client)this;
+            }
+
+            return resource;
+        }
+
+        /// <summary>
+        ///     Execute a request against the EasyPost API.
+        /// </summary>
+        /// <returns>Whether request was successful.</returns>
+        internal async Task<bool> Request(Method method, string url, Dictionary<string, object>? parameters = null, ApiVersion? apiVersion = null)
+        {
+            // Build the request
+            Request request = new Request(url, method, parameters, null, apiVersion);
+            RestRequest restRequest = PrepareRequest(request);
+
+            // Execute the request
+            RestResponse response = await _restClient.ExecuteAsync(restRequest);
+
+            // Return whether the HTTP request produced an error (non-2xx response) or not
```

Comment:
Same `3xx` as above ^

Reply:
Resolved.

Reply:
If so, the comment needs to be updated too.

Reply:
Resolved.